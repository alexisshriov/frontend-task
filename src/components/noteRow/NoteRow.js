import React from 'react';
import Button from 'react-bootstrap/Button';

class NoteRow extends React.Component {
    render() {
        return (
            <div style={styles.noteRow} onClick={() => this.props.selectNote(this.props.note)}>
                <div>
                    <h4>{this.props.note.title}</h4>
                </div>
                <div>
                    <Button className="btn" onClick={() => this.props.deleteNote(this.props.note.id)}>X</Button>
                </div>

            </div>
        );
    }
}

const styles = {
    noteRow: {
        border: '2px solid gray',
        borderRadius: 5,
        padding: 10,
        display: 'flex',
        flex: 1,
        flexDirection: 'row',
        justifyContent: 'space-between'
    }
}

export default NoteRow