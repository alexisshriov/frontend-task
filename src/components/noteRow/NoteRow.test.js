import React from 'react';
import { shallow } from 'enzyme';
import NoteRow from './NoteRow';

describe('NoteList', () => {
  it('should render NoteList correctly', () => {
    const note = {id: 'd39gr', title: 'note title test1', text: 'note text test1'}
    const component = shallow(<NoteRow note = {note}  />);
    expect(component).toMatchSnapshot();
  });
});