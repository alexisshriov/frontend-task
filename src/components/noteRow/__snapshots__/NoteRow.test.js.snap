// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteList should render NoteList correctly 1`] = `
<div
  onClick={[Function]}
  style={
    Object {
      "border": "2px solid gray",
      "borderRadius": 5,
      "display": "flex",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "padding": 10,
    }
  }
>
  <div>
    <h4>
      note title test1
    </h4>
  </div>
  <div>
    <ForwardRef(Bootstrap(Button))
      className="btn"
      onClick={[Function]}
    >
      X
    </ForwardRef(Bootstrap(Button))>
  </div>
</div>
`;
